### PR TITLE
Implement mat4 to isometry3 conversion

### PR DESCRIPTION
**Pull request by thenlevy**
Here is an implementation of #79

### CODE REVIEW

File: src/mat.rs [hunk 1]
```diff
@@ -1356,6 +1356,40 @@ macro_rules! mat4s {
                 (*self * point.into_homogeneous_point()).normalized_homogeneous_point().truncated()
             }
 
+            /// If self represents an `Isometry3` (i.e. self is a product of the from `T * R` where
```
Comment (termhn) [inline src/mat.rs:1359]: This should work as long as self represents an Affine transformation, not just if it's a direct isometry (an affine transform is nonlinear scale, skew, rotation, and translation.

File: src/mat.rs [hunk 2]
```diff
@@ -1356,6 +1356,40 @@ macro_rules! mat4s {
...
+            /// isometry.
+            ///
+            /// If `self` does not represent an isometry, the returned value has undefined
+            /// properties.
+            #[inline]
+            pub fn extract_translation(&self) -> $v3t {
+                self.cols[3].truncated()
+            }
+
+            /// If self represents an `Isometry3` (i.e. self is a product of the from `T * R` where
```
Comment (termhn) [inline src/mat.rs:1370]: Would be more appropriate to just say "if the upper left 3x3 matrix in this matrix is a pure rotation matrix" or something to that effect

**Review by termhn — CHANGES_REQUESTED** [review]
in addition to the two comments, could you also add an entry to the changelog under unreleased? :)

File: src/mat.rs [hunk 3]
```diff
@@ -1367,12 +1365,8 @@ macro_rules! mat4s {
...
             }
 
-            /// If self represents an `Isometry3` (i.e. self is a product of the from `T * R` where
-            /// `T` is a translation and `R` a rotation), return the `rotation` field of the
-            /// isometry.
-            ///
-            /// If `self` does not represent an isometry, the returned value has undefined
-            /// properties.
+            /// If the 3x3 left upper block of `self` is a rotation, return the corresponding
+            /// vector. Otherwise, the returned value is a `Rotor3` with undefined properties.
```
Comment (termhn) [inline src/mat.rs:1367]:
```suggestion
            /// rotor. Otherwise, the returned value is a `Rotor3` with undefined properties.
```

**Review by termhn — CHANGES_REQUESTED** [review]
little typo ;)

**Review by termhn — APPROVED** [review]

**Comment by thenlevy** [comment]
I fixed the typo. I think it is even possible to extract the rotation from an affine transformation (of the form Translation * Rotation * Scaling) because the scaling component can probably be extracted from the trace of the matrix. I'll do the maths to check that and I'll open a new PR if I can make it work

**Pull request merged**